src/controllers/RideOffer.js

```diff
@@ -1,52 +1,125 @@
   try {
-    const data = await driverOfferModel.insertWithReturn(columns, values);
-    res.status(200).json({ msg: data.rows });
+    const data = await offerModel.insertWithReturn(columns, values);
+    return res.status(201)
+      .json({ data: data.rows[0], message: 'offer created successfully', success: true });
   } catch (err) {
-    res.status(400).json({ msg: err.stack });
+    res.status(500).json({ message: err.message });
   }
 };
-
+/**
+ * @description - This method will return a offer for particular driver
+ *
+ * @param {object} req - request
+ *
+ * @param {object} res - response
+ *
+ * @return {object} - it return object
+ */
 export const DriverRideOfferPage = async (req, res) => {
-  const { id } = req.user.driver;
+  const { data: { id } } = req.user;
   try {
-    const data = await driverOfferModel.select('*', `WHERE "driver_id" = '${id}'`);
-    res.status(200).json({ msg: data.rows });
+    const data = await offerModel.select('*', `WHERE "driverId" = '${id}'`);
+    if (data.rowCount === 0) {
+      return res.status(404).json(
+        { data: [], message: 'data does not exist', success: false }
+      );
+    }
+    return res.status(201).json({
+      data: data.rows, success: true
+    });
   } catch (err) {
-    res.status(400).json({ msg: err.stack });
+    res.status(500).json({ message: err.message });
   }
 };
+/**
+ *@description - This method fetch and return all offer object
 
-export const modifyOfferpage = async (req, res) => {
-  const { destination, price } = req.body;
+ * @param {object} req - request
+ *
+ * @param {object} res - response
+ *
+ * @returns {object} - it return a data object and the number of the data
+ */
+export const allOffer = async (req, res) => {
+  let total;
+  try {
+    const data = await offerModel.select('*');
+    total += data.rowCount;
+    if (!data.rowCount) {
+      return res.status(404).json(
+        { data: [], message: 'data does not exist', success: false }
+      );
+    }
+    return res.status(201).json({ data: data.rows, total, success: true, });
+  } catch (err) {
+    res.status(500).json({ message: err.message });
+  }
+};
+/**
+ * @description - The method handle the request for editing a offer
+ *
+ *@return {object} - it return object message
+
+ * @param {object} req - request
+
+ * @param {object} res - response
+ */
+export const editOffers = async (req, res) => {
+  const driverId = req.user.data.id;
   const { id } = req.params;
-  const columns = 'destination, price,';
-  const values = `'${destination}', '${price}'`;
   try {
-    const data = await driverOfferModel.update(columns, values, `WHERE "driver_id" = '${id}'`);
-    console.log(data);
-    res.status(200).json({ msg: data.rows });
+    const data = await offerModel
+      .update(req.body, `WHERE "driverId" = '${driverId}' AND id = '${id}'`);
+    if (!data.rowCount) {
+      return res.status(404).json(
+        { data: [], message: 'data does not exist', success: false }
```

Comment:
What `data`? do you mean `Offers does not exist`?

Reply:
Yes, does it make sense to return "offer does not exist" instead of "data does not exist".  What do you think?